File: dqoai/src/main/java/ai/dqo/rest/models/metadata/CheckSpecBasicModel.java
/*
 * Copyright © 2021 DQO.ai (support@dqo.ai)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ai.dqo.rest.models.metadata;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;

/**
 * CheckSpec basic model that is returned by the REST API.
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@ApiModel(value = "CheckSpecBasicModel", description = "Check spec basic model")
public class CheckSpecBasicModel {
    /**
     * Check name at the leaf level of the check tree.
     */
    @JsonPropertyDescription("Check name")
    private String checkName;

    /**
     * Full check name.
     */
    @JsonPropertyDescription("Full check name")
    private String fullCheckName;

    /**
     * True when the check is a custom check or is customized by the user.
     */
    @JsonPropertyDescription("This check has is a custom check or was customized by the user.")
    public boolean custom;

    /**
     * True when this check is provided with DQO as a built-in check.
     */
    @JsonPropertyDescription("This check is provided with DQO as a built-in check.")
    public boolean builtIn;

    public CheckSpecBasicModel() {
    }
}

File: distribution/python/dqops/client/models/check_spec_basic_model.py
from typing import Any, Dict, List, Type, TypeVar, Union

import attr

from ..types import UNSET, Unset

T = TypeVar("T", bound="CheckSpecBasicModel")


@attr.s(auto_attribs=True)
class CheckSpecBasicModel:
    """Check spec basic model

    Attributes:
        check_name (Union[Unset, str]): Check name
        full_check_name (Union[Unset, str]): Full check name
        custom (Union[Unset, bool]): This check has is a custom check or was customized by the user.
        built_in (Union[Unset, bool]): This check is provided with DQO as a built-in check.
    """

    check_name: Union[Unset, str] = UNSET
    full_check_name: Union[Unset, str] = UNSET
    custom: Union[Unset, bool] = UNSET
    built_in: Union[Unset, bool] = UNSET
    additional_properties: Dict[str, Any] = attr.ib(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        check_name = self.check_name
        full_check_name = self.full_check_name
        custom = self.custom
        built_in = self.built_in

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if check_name is not UNSET:
            field_dict["check_name"] = check_name
        if full_check_name is not UNSET:
            field_dict["full_check_name"] = full_check_name
        if custom is not UNSET:
            field_dict["custom"] = custom
        if built_in is not UNSET:
            field_dict["built_in"] = built_in

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        check_name = d.pop("check_name", UNSET)

        full_check_name = d.pop("full_check_name", UNSET)

        custom = d.pop("custom", UNSET)

        built_in = d.pop("built_in", UNSET)

        check_spec_basic_model = cls(
            check_name=check_name,
            full_check_name=full_check_name,
            custom=custom,
            built_in=built_in,
        )

        check_spec_basic_model.additional_properties = d
        return check_spec_basic_model

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties
